core/src/com/softwarearchitecture/game_server/buttons/Button.java

```diff
@@ -0,0 +1,79 @@
+package com.softwarearchitecture.game_server.buttons;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import com.badlogic.gdx.Gdx;
+import com.badlogic.gdx.graphics.Texture;
+import com.softwarearchitecture.math.Rectangle;
+import com.softwarearchitecture.math.Vector2;
+
+public class Button {
+
+    // should there also be an interface for other stuff that gets drawn on the
```

Comment:
Totaly agree. We should extract needed functionality to the interface as planned. I think that that interface should not expose the SpriteBatch. Only give the user to draw at x, y, width, height, and texture.